### PR TITLE
Format documentation according to TechDocs

### PR DESCRIPTION
**Pull request by nhhagen**
Our TechDocs image expects the documentation to be formatted in a specific way
so that documentation is consistent across our repositories.

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -0,0 +1,291 @@
+# Go Datadog Library
```
Comment (LinMAD) [inline docs/index.md:1]: Is there a practical reason why it collapsed in one long file now?

Comment (nhhagen) [reply docs/index.md:1]: Yes and I should have stated so, there is so little content that I do not think the reader benefits from having to navigate across multiple file.

TechDocs will generate and show a table of content for each file and display it to the right of the content.

I can revert to a multi file setup if that is preferred.

Comment (LinMAD) [reply docs/index.md:1]: It's fine, if you like it more we can do more of it.

**Review by LinMAD — APPROVED** [review]

File: docs/index.md (hunk 1 above)
Comment (nhhagen) [reply docs/index.md:1]: It is a joint effort 😄

**Pull request merged**